### ISSUE TITLE
Check min and max length on all inputs and complain where appropriate.

### ISSUE THREAD

**Issue by rossjones** [datagovuk/publish_data_alpha#281]

**Comment by gtdata** [comment]
As the content is trying to educate users about why they should stick to under 65 characters - is there a way we enforce this? Like Twitter do... @rossjones @maxf @BlancaTortajada

**Comment by maxf** [comment]
@gtdata: yes, it's possible. More specific questions:
1. should we let the user save a draft with a title longer than 65 characters
2. should we let the user publish a dataset with a title longer than 65 characters (or do we just warn them)?
3. Same questions apply to description + what is the limit in characters?

**Comment by rossjones** [comment]
The benefit of imposing restrictions here, is that when users edit existing datasets from the old site, we can enforce the new rules.  So those datasets with very long titles we can force to be shortened.

**Comment by davidread** [comment]
I think it should only be enforced on dataset creation. Occasionally we've had this situation in the past, where someone edits a dataset to change something and it blocks you until you fix something else unrelated. It's just caused confusion and support requests - "why is it complaining, what is wrong with my change?" not understanding it is a different in the error message, even with a pretty clear message and red box drawn around the field with the problem.

**Comment by maxf** [comment]
@davidread if it was enforced all the time, not just on creation, then people editing the dataset would only get error messages on what they just changed, no?

**Comment by davidread** [comment]
Yes for ones created on the new system - that would work. It's the legacy datasets I'm concerned about.

**Comment by maxf** [comment]
Ah, I see. Makes sense for those.
